### PR TITLE
Add access requests to audit events

### LINKED ISSUE #9590: All user-generated events should list the access requests in use
### What
At the time of writing, the `session.start` event contains an array of `access_requests` that were used to gain additional roles at the time that the event was emitted. All events that originate from a user will also contain the impersonator for the user.

We should add the `access_requests` array to all events where it's applicable, not just `session.start` (which only deals with SSH sessions).

### How
After the changes in #9501 the list of access requests that were selected by the user to gain roles should be available in most contexts where the user's identity is available. We could then add a field to `events.UserMetadata`, which is part of almost every event that originate from a user, and add the information whenever we emit an event.

For backwards compatibility, the jsontag for the field should probably match the one used already in the `session.start` event (`access_requests`), but we could also pick a completely new one.

### Why
As the access request workflow allows for privilege escalation, it is important to have as much visibility as possible on what is then done with those escalated privileges.

### PR DESCRIPTION
**Pull request by espadolini**
Closes #9590.

### CODE REVIEW

File: api/types/events/events.proto
Comment (zmb3) [inline api/types/events/events.proto:137]: I'm not familiar - what is this `reserved` magic?

**Review by zmb3 — COMMENTED** [review]
I would not change `e` to point to your branch as part of merging this. Wait until your PR over there is merged and then update.

File: api/types/events/events.proto
Comment (espadolini) [reply api/types/events/events.proto:137]: A `protoc`-enforced reminder to not use that field number or field name in the future; see https://developers.google.com/protocol-buffers/docs/proto3#reserved .

I felt it was appropriate as we're removing a field on what's technically a publicly used message type; perhaps I could move it down and leave a comment about it - the `access_requests` json tag comes from `UserMetadata` now.

Comment (zmb3) [reply api/types/events/events.proto:137]: Gotcha, that makes sense. Have to be careful changing things in events.proto as some of those types are used in session recordings, and you dont want to break parsing an old session.

In this case you should be okay though as I don't think the session start event is ever part of the recorded session.

Comment (espadolini) [reply api/types/events/events.proto:137]: I believe the whole point of reserving fields is that those fields in old versions of the message would just get ignored; this only applies for messages in protobuf form however - in json form the access requests would get parsed as part of `UserMetadata`, which doesn't even lose us any data.

**Review by zmb3 — APPROVED** [review]

**Review by jakule — APPROVED** [review]

**Pull request merged**
